### ISSUE TITLE
Feature: Raffle Draw

### ISSUE THREAD

**Issue by sc-idevops** [antonwilc0x/artmanager#1]
It wouldn't be too hard to modify this program to also be able to pick one of the entries at random for when people do raffles! :3

**Comment by tonytins** [comment]
This is definitely doable but would have to be saved for when I move to a database.

**Comment by tonytins** [comment]
Okay, I've been thinking this over and it is possible to create a raffle system with the current architecture. It'll be accessed via a subcommand since it differs from normal routine. ``artm raffle --slots 5 --tickets 20 --ych "Night Sky"``. The slots and tickets are max that are allowed and will generate a number based on those limitations. So the ouput would be:
```json
{
  "slot": "2",
  "ticket": "15",
  "ych": "Night Sky"
}
```
Of course, this is very basic and a lot of room for error but it's a start. It's based on [this](https://www.furaffinity.net/view/28299665/) YCH raffle.

**Comment by sc-idevops** [comment]
ah I see how this raffle is set up. Users are self assigned a number on the order they are commented in. I was originally thinking about a twitter raffle where you can plug in someone's name and data in and then select one of those, but I think this implementation is a lot easier and requires less input. (all it is really is, is a RNG)

**Comment by tonytins** [comment]
Yes. I do like both the idea of a raffle and your Twitter-based approach. It's just the current architecture doesn't really allow for it, at the moment. I figured this would be the best compromise in the meantime.

**Comment by antonwilc0x** [comment]
Here is a possible system. 

``raffle`` or ``raf`` would be a subcommand tied to the ych one. It would be the two step process. The first sets name and maximum slots and tickets. ``artm ych raf --art <art> --tkts <tickets> --slots <slots>``.

```json
{
  "id": "741b4f90-09f4-4122-bbbf-3096957b9787",
  "art": "Night sky",
  "slots": "2",
  "tickets": "15"
}
```

The next step takes the drawer file and generates a raffle variant of the YCH format so it can be edited by the client. ``artm ych raf --file <file>``.

```json
{
  "id": "f855b7ac-af3d-4594-88a5-0279c5278de7",
  "art": "Night sky",
  "slot": "1",
  "ticket": "3",
  "contact": "",
  "payment": "",
  "reference": ""
}
```
What I need to do first is provide safe handling of duplicate files. Right now the application crashes if one is found. This would be made even complicated with raffles. I've held off on doing this in order to finish building and testing the core functionality.

**Comment by tonytins** [comment]
The architecture was completely rewritten starting with 0.4. I think this feature may finally become more of a reality.
